### DISCUSSION TITLE
How to reduce image size

### DISCUSSION THREAD

**Discussion by satimis** [ImageMagick/ImageMagick#5222] [Digital Image Processing]
**How to reduce image size**

Original size: 4092(w) x 1960(h)

Final size: 2527(w) x 1815(h)

Please advise what command shall I use?  Thanks

Regards

**Comment by snibgo** [comment]
```
magick in.ext -resize 2527x1815! out.ext
```

**Comment by satimis** [comment]
Hi snibgo,

I got it.  Thanks.

A further question, how to crop an image?
magick input.jpg -crop 2527x1815+300+20 output.jpg
It doesn't work.

Also;
magick input.jpg -crop 2527x1815+300+20 +repage output.jpg
It does't work

Regards

**Reply by fmw42** [reply]
What do you want to do? Can you provide examples?  **In what way does it not work?**

Please always provide your IM version and platform when asking questions.

**Comment by satimis** [comment]
Hi fmw42
I did cropping on GIMP.  Please refer to attached images.  But I couldn't did it on IM

![original30](https://user-images.githubusercontent.com/30752517/173621524-ead2000c-5aae-4d42-921b-a0d837ded9f9.jpg)
![cropped_gimp_30](https://user-images.githubusercontent.com/30752517/173621574-5f16dd0a-b7d3-4689-a594-415a5debf079.jpg)
![original_crop_im_30](https://user-images.githubusercontent.com/30752517/173621651-1d28c816-fb64-4701-8d9e-b7b72540e21c.jpg)

**Comment by snibgo** [comment]
@satimis: please show actual files, not screenshots. We need to reproduce your problem.

> magick input.jpg -crop 2527x1815+300+20 +repage output.jpg
It does't work

How do you know it doesn't work? What happens instead? Do you get any error messages?

**Reply by satimis** [reply]
No error message.  But the output image is photo 3, not similar to photo 2.
What command shall I run to produce the output image similar to photo 2?

**Comment by fmw42** [comment]
What numbers did you use in GIMP for your cropping or did you crop visually without exact numbers? What do the number represent in GIMP compared to the numbers in Imagemagick?

**Reply by satimis** [reply]
I crop visually without exact numbers.
Tools - Transform Tools -> Crop

**Comment by fmw42** [comment]
How did you crop in GIMP compared to Imagemagick. They crop totally differently. As far as I know in GIMP you mark the top-left and bottom-right corners using the rulers doing it visually.  In Imagemagick the numbers are WxH+X+Y. So that is the top-left corner from +X+Y and the width and height. So these numbers are not the same as from GIMP. To convert GIMP values to Imagemagick your need subtract the two corners to get the width and height and then use the top left corner and the width and height, not the top-left and bottom-right corner values.

**Comment by fmw42** [comment]
I get the same results in IM and GIMP. I think you have your WxH swapped with +X+Y.  Imagemagick crop is WxH+X+Y.  Where +X+Y is the top left corner position and WxH is the width and height sizes.

Here is a crop in GIMP. Note the values.

Input:

![barn](https://user-images.githubusercontent.com/13458218/173630961-f6a5981d-b84c-4dec-b78b-64ff7477c2ad.jpg)

GIMP (Note the values):

![Screen Shot 2022-06-14 at 9 38 25 AM](https://user-images.githubusercontent.com/13458218/173630993-fc710e5b-a40b-4e0c-a031-50ad43373fae.png)


Imagemagick (Note the values and order):

`convert barn.jpg -crop 199x150+102+100 +repage x.jpg`

![x](https://user-images.githubusercontent.com/13458218/173631160-9e5c6f4f-71e2-4fe0-b376-070762f97962.jpg)

In GIMP, the position values correspond to +X+Y in Imagemagick. The size values in GIMP correspond to WxH in Imagemagick.

**Comment by snibgo** [comment]
@satimis: The first image you show is:
```
magick 173621524-ead2000c-5aae-4d42-921b-a0d837ded9f9.jpg info:

173621524-ead2000c-5aae-4d42-921b-a0d837ded9f9.jpg JPEG 588x1210 588x1210+0+0 8-bit sRGB 211729B 0.000u 0:00.002
```
It is 588x1210, so taller than it is wide. It also has an orientation setting.

I suggest using "-auto-orient" after reading the input.
